### DISCUSSION TITLE
[Enhancement] [mention] Mention 组件希望可以抛出dropdownVisible属性

### DISCUSSION THREAD

**Discussion by SorrowX** [element-plus/element-plus#18457] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `mention`

### Additional comments

需求：
基于Mention组件封装发送框时，需要按enter直接发送，shift+enter换行。
这时需要监听onKeydown事件，回车时来按需阻止浏览器默认事件，到这里一切正常。
可是我们的Mention组件当出现提及弹窗时，用户进行键盘上下按键选择完后，回车选中后，这里我需要知道Mention组件的弹窗状态，如果处于打开，我监听的onKeydown事件函数直接return即可，目前拿不到该状态，导致我们用户层在封装时比较麻烦。
希望抛出该属性

**Comment by warmthsea** [comment]
回车后 `dropdownVisible` 的值似乎是一定为 false 的

**Reply by SorrowX** [reply]
看mention组件的实现确实会把visible重置为false,导致计算属性dropdownVisible为false.

**Reply by SorrowX** [reply]
![image](https://github.com/user-attachments/assets/e45a81e4-1960-4698-bac1-7a1e00c45699)
![image](https://github.com/user-attachments/assets/ba9473d0-3be5-4dc1-8c3b-a1593ebe16ad)
我写了个demo,在外部监听的事件会优先触发，然后执行内部的事件回调，目前这种场景对我来说是正常的

**Reply by SorrowX** [reply]
![image](https://github.com/user-attachments/assets/96e8035e-5dcf-4fc3-9e1d-6c2f6128efb7)
上述的执行顺序也很好理解，我们写在模版上的事件，编译后通过addEventListener优先注册外部回调然后就是其mention组件中的el-input keydown事件。当回车tooltip弹窗时，这时dropdownVisible还是true,因为先执行外部的keydown事件回调，用户拿到的还是dropdownVisible为true,然后执行内部的 keydown事件回调，这是才是false。
目前这种场景对我来说是极好的。
如果用户在外侧的 keydown事件想拿到dropdownVisible为false的值，可以在下一次事件循环就可以拿到，我觉得按需选择即可

**Reply by SorrowX** [reply]
当然上面所有的解释都是我希望mention组件可以抛出一个tooltip显隐的状态。
当时我有3种方案。
第一种就是上述的直接抛出
第二种就是在watch dropdownVisible值的改变 发射emit事件
第三种就是在每个 visible赋值时 发射emit事件

我觉得代码简洁且改动极少也不会产生其他副作用就选了第一种。

**Comment by warmthsea** [comment]
第一次 `Enter` 时候，是选择
第二次 `Enter` 时候，你想发送消息做其他操作 对么

![screenshots](https://github.com/user-attachments/assets/7849bb7e-f589-47f8-89e6-780d6a72cc53)

**Reply by SorrowX** [reply]
是的，我想基于该组件封装个发送框

**Reply by SorrowX** [reply]
![image](https://github.com/user-attachments/assets/1b906979-9e66-4717-8e9c-fa3ac918e119)
我目前的做法是监听了2个事件来模拟了dropdownVisible状态值，代码就变得繁琐，所以希望官方提供个api
